### PR TITLE
docs(overlay): expand overview for overlay

### PR DESCRIPTION
**Pull request by jelbourn**

### CODE REVIEW

File: src/cdk/overlay/overlay.md [hunk 1]
```diff
@@ -1,3 +1,88 @@
...
 
-A service used to manage overlays.
\ No newline at end of file
+The `Overlay` service provides a way to open floating panels on the screen.
+
+### Creating overlays
+Calling `overlay.create()` will return an `OverlayRef` instance. This instance is a handle for
+managing that specific overlay.
+
+The `OverlayRef` _is_ a `PortalHost`- once creates, content can be added by attaching a `Portal`.
```
Comment (crisbeto) [inline src/cdk/overlay/overlay.md]: once creates -> once created

File: src/cdk/overlay/overlay.md [hunk 2]
```diff
@@ -1,3 +1,88 @@
...
+See the documentation on portals for further information.
```
Comment (crisbeto) [inline src/cdk/overlay/overlay.md]: Link to the portal docs (if we have any)?

File: src/cdk/overlay/overlay.md [hunk 3]
```diff
@@ -1,3 +1,88 @@
...
+The `scrollStrategy` configuration option determines how the overlay will react to scrolling outside
+the overlay element. There are four scroll strategies available as part of the library.
+
+`NoopScrollStrategy` is the default option. This strategy does nothing.
+
+`CloseScrollStrategy` will automatically close the overlay when scrolling occurs.
+
+`BlockScrollStrategy` will block page scrolling while the overlay is open. Note that some
+applications may implement special or customized page scrolling; if the `BlockScrollStrategy`
+conflicts with this kind of situation, it can overriden by re-providing `BlockScrollStrategy` with
```
Comment (kara) [inline src/cdk/overlay/overlay.md]: can overridden -> can be overridden

File: src/cdk/overlay/overlay.md [hunk 4]
```diff
@@ -1,3 +1,88 @@
...
+a custom implementation.
+
+`RepositionScrollStrategy` will re-position the overlay element on scroll. Note that this will have
+some performance impact on scrolling- users should weigh this cost in the context each specific
```
Comment (kara) [inline src/cdk/overlay/overlay.md]: in the context each -> in the context of each

File: src/cdk/overlay/overlay.md (hunk 1 above)
Comment (jelbourn) [reply src/cdk/overlay/overlay.md]: Done

File: src/cdk/overlay/overlay.md (hunk 2 above)
Comment (jelbourn) [reply src/cdk/overlay/overlay.md]: They don't exist yet; I'm planning to do a pass and ass cross-references once I have more of the content authored

File: src/cdk/overlay/overlay.md (hunk 3 above)
Comment (jelbourn) [reply src/cdk/overlay/overlay.md]: Done

File: src/cdk/overlay/overlay.md (hunk 4 above)
Comment (jelbourn) [reply src/cdk/overlay/overlay.md]: Done

File: src/cdk/overlay/overlay.md [hunk 5]
```diff
@@ -1,3 +1,88 @@
...
+
+### The overlay container
+The `OverlayContainer` provides a handle to the container element in which all individual overlay
+elements are rendered. By default, the overlay container is appended directly to the document body
+so that an overlay is never clipped by an `overflow: hidden` parent.
+
+#### Full-screen overlays
+The `FullscreenOverlayContainer` is an alternative to `OverlayContainer` that supports correct
+displaying of overlay elements in
+[fullscreen mode][https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullScreen].
```
Comment (willshowell) [inline src/cdk/overlay/overlay.md]: Link syntax

Comment (jelbourn) [reply src/cdk/overlay/overlay.md]: Whoops, good catch

**Pull request merged**